backend/lm_backend/api/routes/bookings.py

```diff
@@ -26,8 +28,61 @@ async def create_booking(
     booking: BookingCreateSchema,
     db_session: AsyncSession = Depends(get_session),
 ):
-    """Create a new booking."""
-    return await crud_booking.create(db_session=db_session, obj=booking)
+    """Create a new booking.
+
+    Use a somewhat complex logic to check if a booking can be made and insert it in one query. In this case,
+    implement the `INSERT columns FROM select_query WHERE EXISTS (SELECT subquery)` SQL paradigm using
+    SQLAlchemy query API. A  booking will only be created if there are enough licenses available to make the
+    booking.
+    """
+
+    # CTE used to provide the request values to the insert/select query.
+    vars_cte = select(
+        literal(booking.feature_id).label("feature_id"),
+        literal(booking.job_id).label("job_id"),
+        literal(booking.quantity).label("quantity"),
+    ).cte("vars")
+
+    # Subquery to determine if there are enough licenses to be booked
+    exists_subquery = (
+        select(Feature.id)
+        .select_from(Feature)
+        .join(Booking, and_(Feature.id == Booking.feature_id, Booking.feature_id == booking.feature_id))
+        .join(Inventory, Feature.id == Inventory.feature_id)
```

Comment:
I believe one `join` inside the `select_from` and then `where` makes it easier to read and understand